### PR TITLE
Wanchain compatibility changes

### PR DESCRIPTION
**Pull request by MoMannn**
DONE:
- Spliting deploy gateway to proxy and gateway contract to lower contract gas consumption.
- Changing AddressUtils to work with byzantanium EVM (updating isContract function).
- Add contract README descriptions for wanchain build (use byzantanium not constantinople)
- Update wanchain generic provider to have different default contract links.

### CODE REVIEW

**Comment by fulldecent** [comment]
Can you please explain "Spliting deploy gateway to proxy and gateway contract to lower contract gas consumption."

This is scaring me.

**Comment by MoMannn** [comment]
> Can you please explain "Spliting deploy gateway to proxy and gateway contract to lower contract gas consumption."
> 
> This is scaring me.

So XcertDeployGateway uses 5 500 000 gas to deploy. Since it contains the whole xcert + logic etc. Wamchain block limit is 4 700 000 so it cannot be deployed there. For this reason I split the XcertDeployGateway to XcertDeployProxy (which only does deploy) and XcertDeployGateway - which contains the logic. To ne consistent everywhere I did this with TokenDeployGateway aswell.

**Comment by xpepermint** [comment]
@fulldecent all good to merge?

File: packages/0xcert-ethereum-proxy-contracts/src/contracts/xcert-custom.sol [hunk 1]
```diff
@@ -36,8 +35,7 @@ contract XcertCustom is XcertToken {
     string memory _uriPostfix,
     bytes32 _schemaId,
     bytes4[] memory _capabilities,
-    address _owner,
-    address[5] memory _proxies
+    address[6] memory _addresses
```
Comment (fulldecent) [inline packages/0xcert-ethereum-proxy-contracts/src/contracts/xcert-custom.sol]: This variable naming is opaque. Can the separate parameters be broken out and named?

**Comment by fulldecent** [comment]
Cool. I didn't realize we were up to _that_ much gas already. Looks good. Minor comment above.

File: packages/0xcert-ethereum-proxy-contracts/src/contracts/xcert-custom.sol (hunk 1 above)
Comment (MoMannn) [reply packages/0xcert-ethereum-proxy-contracts/src/contracts/xcert-custom.sol]: They where in the beginning but it broke code depth complexity. Well actual first every proxy was standalone, then owner and proxies now for the best gas performance everything is in the same array.

**Pull request merged**